invenio_users_resources/resources/groups/resource.py

```diff
@@ -52,25 +52,31 @@ def search(self):
     @response_handler()
     def read(self):
         """Read a group."""
-        item = self.service.read(
-            id_=resource_requestctx.view_args["id"],
-            identity=g.identity,
-        )
-        return item.to_dict(), 200
+        try:
+            item = self.service.read(
+                id_=resource_requestctx.view_args["id"],
+                identity=g.identity,
+            )
+            return item.to_dict(), 200
+        except LookupError:
+            abort(404)
 
     @request_view_args
     def avatar(self):
         """Get a groups's avatar."""
-        avatar = self.service.read_avatar(
-            name_=resource_requestctx.view_args["id"],
-            identity=g.identity,
-        )
-        return send_file(
-            avatar.bytes_io,
-            mimetype=avatar.mimetype,
-            as_attachment=False,
-            download_name=avatar.name,
-            etag=avatar.etag,
-            last_modified=avatar.last_modified,
-            max_age=86400 * 7,
-        )
+        try:
+            avatar = self.service.read_avatar(
+                name_=resource_requestctx.view_args["id"],
+                identity=g.identity,
+            )
+            return send_file(
+                avatar.bytes_io,
+                mimetype=avatar.mimetype,
+                as_attachment=False,
+                download_name=avatar.name,
+                etag=avatar.etag,
+                last_modified=avatar.last_modified,
+                max_age=86400 * 7,
+            )
+        except AttributeError:
```

Comment:
this error block seems a little bit too broad, it will catch all the AttributeError exceptions (not only the one we want to address), potentially hiding other bugs. Where exactly is the AtrributeError thrown ? I would try to handle it there

Reply:
I agree, there is a missing check [here](https://github.com/inveniosoftware/invenio-users-resources/blob/master/invenio_users_resources/services/groups/service.py#L41) I can raise a LookupError there for read_avatar to be consistent with def read, and except a LookupError def avatar, will that be an acceptable solution?